### PR TITLE
[FIX] l10n_in_hr_holidays: fixed half day sandwich leave duration

### PR DESCRIPTION
**Pull request by yava-odoo**
**Steps:**
1. Install the `l10n_in_hr_holidays` module and create an Indian company
2. Create a time-off type that has `half-days` as the request unit and set "sandwich leave" to true
3. Create a leave request from December 12 to December 15, 2025 it should display a duration of 2 days instead of 4

**Cause:**
The filter was removing leaves that had `half-day` set as the request unit

**Fix:**
The half-day filter is removed.
In `_l10n_in_is_full_day_request` method we will check for actual half-day leave If the `request_date_from_period` and `request_date_to_period` are the same, then it is not considered a full-day leave, and it will not be calculated as a sandwich leave.

Task-5427415

### CODE REVIEW

File: addons/l10n_in_hr_holidays/models/hr_leave.py [hunk 1]
```diff
@@ -31,11 +31,13 @@ def _l10n_in_get_default_leave_hours(self):
         data = work_data.get(self.employee_id.id)
         return data.get("hours", 0.0) if data else 0.0
 
-    def _l10n_in_is_full_day_request(self, hours=None, default_hours=None):
+    def _l10n_in_is_full_day_request(self, hours=None):
```
Comment (amah-odoo) [inline addons/l10n_in_hr_holidays/models/hr_leave.py]: we should not change method signature in stable 🤔

File: addons/l10n_in_hr_holidays/tests/test_sandwich_leave.py [hunk 2]
```diff
@@ -329,6 +329,60 @@ def test_sandwich_for_leave_type_hours(self):
...
+            'name': 'Test Leave',
+            'employee_id': self.rahul_emp.id,
+            'holiday_status_id': self.leave_type_half_day.id,
+            'request_date_from': "2025-12-12",
+            'request_date_to': "2025-12-15",
+            'request_date_from_period': 'pm',
+            'request_date_to_period': 'am',
+        })
+        self.assertTrue(holiday_leave.l10n_in_contains_sandwich_leaves)
+        self.assertEqual(holiday_leave.number_of_days, 4)
```
Comment (amah-odoo) [inline addons/l10n_in_hr_holidays/tests/test_sandwich_leave.py:366]: I think this should be 3 days 👀 
0.5 day 12th after noon
2 days 13th & 14th  full days
0.5 day 15th morning

File: addons/l10n_in_hr_holidays/tests/test_sandwich_leave.py [hunk 3]
```diff
@@ -329,6 +329,60 @@ def test_sandwich_for_leave_type_hours(self):
...
+        })
+        self.assertFalse(before_holiday_leave.l10n_in_contains_sandwich_leaves)
+        self.assertTrue(after_holiday_leave.l10n_in_contains_sandwich_leaves)
+        self.assertEqual(after_holiday_leave.number_of_days, 3)
+
+        after_holiday_leave.write({
+            'request_date_from_period': 'pm',
+            'request_date_to_period': 'pm',
+        })
+        self.env['hr.leave'].invalidate_model()
```
Comment (amah-odoo) [inline addons/l10n_in_hr_holidays/tests/test_sandwich_leave.py]: I think we do not need this but we should trigger the check for the `l10n_in_contains_sandwich_leaves` automatically when we change the period as we do when we change the dates 👀

**Review by amah-odoo — CHANGES_REQUESTED** [review]
Thanks for your work

File: addons/l10n_in_hr_holidays/models/hr_leave.py (hunk 1 above)
Comment (yava-odoo) [reply addons/l10n_in_hr_holidays/models/hr_leave.py]: I forgot to remove this diff, Earlier it was in the master 🙈

File: addons/l10n_in_hr_holidays/tests/test_sandwich_leave.py (hunk 2 above)
Comment (yava-odoo) [reply addons/l10n_in_hr_holidays/tests/test_sandwich_leave.py:366]: That was one edge case that I missed

We are not considering the half-day start or end leave in the sandwich leave calculations
So the duration for this case will be 1 day

File: addons/l10n_in_hr_holidays/tests/test_sandwich_leave.py (hunk 3 above)
Comment (yava-odoo) [reply addons/l10n_in_hr_holidays/tests/test_sandwich_leave.py]: Right, I just need to read the `number_of_days` first, this will recompute the whole thing
Then we can check for the `l10n_in_contains_sandwich_leaves`

**Review by amah-odoo — APPROVED** [review]
Thanks for your work 💯
